### ISSUE TITLE
wrong loss function in examples for VI reparameterization layers

### ISSUE THREAD

**Issue by hongkahjun** [tensorflow/probability#1335]
In example for https://www.tensorflow.org/probability/api_docs/python/tfp/layers/DenseReparameterization, it states

```
import tensorflow as tf
import tensorflow_probability as tfp

model = tf.keras.Sequential([
    tfp.layers.DenseReparameterization(512, activation=tf.nn.relu),
    tfp.layers.DenseReparameterization(10),
])

logits = model(features)
neg_log_likelihood = tf.nn.softmax_cross_entropy_with_logits(
    labels=labels, logits=logits)
kl = sum(model.losses)
loss = neg_log_likelihood + kl
train_op = tf.train.AdamOptimizer().minimize(loss)
```

However, the KL loss is already incorporated in the call functions of any child that inherits _DenseVariational

https://github.com/tensorflow/probability/blob/4683836be3701e75f459494bd3de12edb7560a2d/tensorflow_probability/python/layers/dense_variational.py#L308

Thus the examples of any layer that inherits _DenseVariational and also DenseVariational should exclude kl in the loss functions. ie.

```
import tensorflow as tf
import tensorflow_probability as tfp

model = tf.keras.Sequential([
    tfp.layers.DenseReparameterization(512, activation=tf.nn.relu),
    tfp.layers.DenseReparameterization(10),
])

logits = model(features)
neg_log_likelihood = tf.nn.softmax_cross_entropy_with_logits(
    labels=labels, logits=logits)
loss = neg_log_likelihood
train_op = tf.train.AdamOptimizer().minimize(loss)
```

Please feel free to correct me if I am wrong.
